app/experimenter/experiments/models/nimbus.py

```diff
@@ -458,53 +458,51 @@ def __str__(self):  # pragma: no cover
 
 
 class NimbusChangeLogManager(models.Manager):
+    def latest_change(self):
+        return self.all().order_by("-changed_on").first()
+
     def latest_review_request(self):
         return (
             self.all()
             .filter(
-                Q(old_status=NimbusExperiment.Status.DRAFT)
-                | Q(old_status=NimbusExperiment.Status.PREVIEW),
+                Q(old_status=F("new_status"))
+                | Q(
+                    old_status=NimbusExperiment.Status.PREVIEW,
+                    new_status=NimbusExperiment.Status.DRAFT,
+                )
             )
             .filter(
                 old_publish_status=NimbusExperiment.PublishStatus.IDLE,
-                new_status=NimbusExperiment.Status.DRAFT,
                 new_publish_status=NimbusExperiment.PublishStatus.REVIEW,
             )
             .order_by("-changed_on")
         ).first()
 
     def latest_rejection(self):
-        return (
-            self.all()
-            .filter(
-                Q(old_publish_status=NimbusExperiment.PublishStatus.REVIEW)
-                | Q(old_publish_status=NimbusExperiment.PublishStatus.WAITING),
-            )
-            .filter(
-                new_publish_status=NimbusExperiment.PublishStatus.IDLE,
+        change = self.latest_change()
+        if (
+            change
+            and change.old_status == change.new_status
+            and change.old_publish_status
+            in (
+                NimbusExperiment.PublishStatus.REVIEW,
+                NimbusExperiment.PublishStatus.WAITING,
             )
-            .order_by("-changed_on")
-        ).first()
+            and change.new_publish_status == NimbusExperiment.PublishStatus.IDLE
+        ):
+            return change
+        return None
```

Comment:
Functions return `None` by default, so you can make this explicit or just omit.  I always omit, the only time I've seen anything complain about it is mypy which makes you always explicitly declare a return for all branches.